test/converge.js

```diff
@@ -3,42 +3,38 @@ var assert = require('assert');
 var R = require('..');
 
 
+//  square :: Number -> Number
+var square = function(n) { return n * n; };
+
+
 describe('converge', function() {
-  var mult = function(a, b) {return a * b;};
-
-  var f1 = R.converge(mult,
-                      function(a) { return a; },
-                      function(a) { return a; });
-  var f2 = R.converge(mult,
-                      function(a) { return a; },
-                      function(a, b) { return b; });
-  var f3 = R.converge(mult,
-                      function(a) { return a; },
-                      function(a, b, c) { return c; });
-
-  it('passes the results of applying the arguments individually to two separate functions into a single one', function() {
-    assert.strictEqual(R.converge(mult, R.add(1), R.add(3))(2), 15); // mult(add1(2), add3(2)) = mult(3, 5) = 3 * 15;
+
+  it('is a ternary function', function() {
+    assert.strictEqual(typeof R.converge, 'function');
+    assert.strictEqual(R.converge.length, 3);
   });
 
-  it('returns a function with the length of the "longest" argument', function() {
-    assert.strictEqual(f1.length, 1);
-    assert.strictEqual(f2.length, 2);
-    assert.strictEqual(f3.length, 3);
+  it('diverges and converges', function() {
+    assert.strictEqual(R.converge(R.subtract, square, R.inc)(10), (10 * 10) - (10 + 1));
   });
 
   it('passes context to its functions', function() {
-    var a = function(x) { return this.f1(x); };
-    var b = function(x) { return this.f2(x); };
-    var c = function(x, y) { return this.f3(x, y); };
-    var d = R.converge(c, a, b);
-    var context = {f1: R.add(1), f2: R.add(2), f3: R.add};
-    assert.equal(a.call(context, 1), 2);
-    assert.equal(b.call(context, 1), 3);
-    assert.equal(d.call(context, 1), 5);
+    var c = function(a, b) { return this.f(a, b); };
+    var f = function(a) { return this.x + a; };
+    var g = function(a) { return this.x - a; };
+    var ctx = {
+      f: function(a, b) { return Math.pow(a - b, this.x); },
+      x: 3
+    };
+
+    assert.strictEqual(R.converge(c, f, g).call(ctx, 0), 0);  // Math.pow((3 + 0) - (3 - 0), 3)
+    assert.strictEqual(R.converge(c, f, g).call(ctx, 1), 8);  // Math.pow((3 + 1) - (3 - 1), 3)
+    assert.strictEqual(R.converge(c, f, g).call(ctx, 2), 64); // Math.pow((3 + 2) - (3 - 2), 3)
   });
 
-  it('returns a curried function', function() {
-    assert.strictEqual(f2(6)(7), 42);
-    assert.strictEqual(f3(R.__).length, 3);
+  it('is curried', function() {
+    assert.strictEqual(R.converge(R.subtract).length, 2);
+    assert.strictEqual(R.converge(R.subtract)(square).length, 1);
   });
```

Comment:
This is also a substantive change in the tests.  The old tests demonstrated that `converge(f, g, h)` is curried.  Yours demonstrates that `converge` is curried.  Should we have both?

Certainly the whole show-every-function-is-curried routine has long since gotten old.  But the old unit test was something different.